assessment/libs/complex.php

```diff
@@ -0,0 +1,636 @@
+    $tht=0;
+    $counter=count($num);
+    for ($i=0; $i < $counter; $i++){
+        
+        $re=$num[$i][0];
+        $im=$num[$i][1];
+        $sq=$re**2+$im**2;
+        $r1= sqrt($sq);
+     
+        $th1=asin(abs($im/$r1));
+
+        if ($re>=0 && $im>=0){
+                $theta=$th1;
+        }    
+            else if($re<0 && $im<0){
+                $theta=pi() +$th1;
+            }
+            else if($re<0 && $im>=0){
+                $theta=pi() -$th1;
+            }
+            else {
+                $theta=2*pi() -$th1;
+            }
+    
+        $rt=$rt*$r1;
+        $tht+=$theta;
+    }
+
+    $re_p=$rt*cos($tht);
+    $im_p=$rt*sin($tht); 
+    
+    $st= array([round($re_p, $roundto) , round($im_p,$roundto)]);    
+        
+    return $st;
+}
+
+//-------------------------------------------------Complex power--------------------------------------------
+// Function: cx_pow(num, exp, [roundto=12])
+// Returns the power of a complex number in standard form as a paired value in an array. 
+//
+// Parameters:
+// num: An array of real and imaginary parts of a complex number given in square brackets: num = array([Re, Im]).
+// exp: exponent to which the complex number is raised.
+// roundto: Optional - number of decimal places to which values should be rounded off; 
+//          default is 12 decimal places.
+// 
+// Returns:
+// The real and imaginary parts of the power as a paired value in an array: array([Re, Im]).
+
+function cx_pow(array $num, $pow, int $roundto=12) {
+    if (count($num)!=1) { echo "cx_pow expects 1 complex number as an input"; return "";}
+    $r1= cx_modul($num);
+    $th1=cx_arg($num);
+    $r=$r1**$pow;
+    $tht=$th1*$pow;
+    
+    $re=$r*cos($tht);
+    $im=$r*sin($tht);  
+        
+    $st= array([round($re,$roundto) , round($im,$roundto)]);   
+        
+    return $st;
+}
+
+//-----------------------------------------------Complex nth Roots--------------------------------------------
+// Function: cx_root(num, n, [roundto=12])
+// Returns an array of the nth roots of a complex number in standard form.  
+//
+// Parameters:
+// num: An array of real and imaginary parts of a complex number given in square brackets: num = array([Re, Im]).
+// n: The nth root. For example, for the cubic roots n=3.
+// roundto: Optional - number of decimal places to which values should be rounded off; 
+//          default is 12 decimal places.
+// 
+// Returns:
+// An array of a paired value of real and imaginary parts of the nth roots. 
+
+function cx_root(array $num, int $root, int $roundto=12) {
+    if (count($num)!=1) { echo "cx_root expects 1 complex number as an input"; return "";}
+    $r1= cx_modul($num);
+    $th1= cx_arg($num);
+    $r=$r1**(1/$root);
+    $A=array();
+    
+    for ($k=0; $k<$root; $k++){
+        
+        $tht=$th1+2*pi()*$k/$root;
+        $re_f=$r*cos($tht);
+        $im_f=$r*sin($tht);
+        $A[$k]=[round($re_f,$roundto) , round($im_f,$roundto)];
+    }
+    return $A;
```

Comment:
Here the array of arrays makes sense, since there are multiple values being returned.